specs/agents/tracing-instrumentation-messaging.md

```diff
@@ -0,0 +1,103 @@
+- **`context.message.age.ms`**: optional for message/record receivers only (transactions or spans). 
+A numeric field indicating the message's age in milliseconds. Relevant for transactions and 
+`receive` spans that receive valid messages. There is no accurate definition as to how this is calculated. If the messaging framework 
+provides a timestamp for the message- agents may use it. Otherwise, the sending agent can add a timestamp _indicated as milliseconds since 
+epoch UTC_ to the message's metadata to be retrieved by the receiving agent. If a timestamp is not available- agents should omit this field. 
+Clock skews between agents are ignored, unless the calculated age (receive-timestamp minus send-timestamp) is negative, in which case the 
+agent should report 0 for this field.
+- **`context.message.routing-key`**: optional. Use only where relevant. Currently only RabbitMQ.
+
+#### Transaction context fields
+
+- **`context.message.body`**: similar to HTTP requests' `context.request.body`- only fill in messaging-related **transactions** (ie 
+incoming messages creating a transaction) and not for outgoing messaging spans. 
+   - Capture only when `ELASTIC_APM_CAPTURE_BODY` config is set to `true`.
+   - Only capture UTF-8 encoded message bodies.
+   - Limit size to 10000 characters. If longer than this size, trim to 9999 and append with ellipsis
+- **`context.message.headers`**: similar to HTTP requests' `context.request.headers`- only fill in messaging-related **transactions**.
+   - Capture only when `ELASTIC_APM_CAPTURE_HEADERS` config is set to `true`.
+   - Sanitize headers with keys configured through `ELASTIC_APM_SANITIZE_FIELD_NAMES`.
+   - Intake: key-value pairs, same like `context.request.headers`.
+
+#### Span context fields
+
+- **`context.destination.address`**: optional. Not available in some cases. Only set if the actual connection is available.
+- **`context.destination.port`**: optional. Not available in some cases. Only set if the actual connection is available.
+- **`context.destination.service.name`**: mandatory. Use the framework name in lowercase, e.g. `kafka`, `rabbitmq`.
+- **`context.destination.service.resource`**: mandatory. Use the framework name in lowercase. Wherever the queue/topic/exchange name is
+ available, append it with a leading `/`, for example: `kafka/myTopic`, `rabbitmq/myExchange`, `rabbitmq`.
+- **`context.destination.service.type`**: mandatory. Use `messaging`.
+
+### `ELASTIC_APM_IGNORE_MESSAGE_QUEUES` configuration
+
+Used to filter out specific messaging queues/topics from being traced.
+
+This property should be set to a list containing one or more strings. When set, sends-to and receives-from the specified 
+queues/topic will be ignored.
+
+|                |   |
+|----------------|---|
+| Type           | `List<`[`WildcardMatcher`](../../tests/agents/json-specs/wildcard_matcher_tests.json)`>` |
+| Default        | empty |
+| Dynamic        | `true` |
+| Central config | `false` |
```

Comment:
any particular reason not adding it to central config?

Reply:
No, at some point we should. 
This reflects the current status, where it is not there.
Once it becomes a priority and we add it, we should change here as well I guess.

Reply:
I think the spec should reflect the desired status. If there's a discrepancy between an agent and the spec, there should be an issue in the agent.

Reply:
This is not only a discrepancy between the spec and the agent, but with Kibana as well. I don't mind changing it here, but it may continue to be a discrepancy for a long time, until we get to that. 
What do you say I will open an issue in the APM repo for adding this central config option and I will add a checkbox for updating this part of the spec?

Reply:
For someone implementing this spec, this reads as don't implement central config which is the wrong message I think. 

>  it may continue to be a discrepancy for a long time, until we get to that

I think that's fine as long as we have an open issue for that.